### PR TITLE
Updated icons to use Font Awesome

### PR DESCRIPTION
**Pull request by InsaneSplash**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/) - please do NOT submit a pull request unless you have (signing the agreement in the same pull request is fine). Your commit message for signing the agreement must appear as per the docs.
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

### CODE REVIEW

**Comment by murrant** [comment]
Like this overall, but some of the replacements are a little too generic.  Like specifically memory->cog.  I think it is ok to keep the sensor icons as regular images for now.

File: html/includes/table/devices.inc.php [hunk 1]
```diff
@@ -203,11 +203,11 @@
             $col_port = '';
         }
         if ($port_count) {
-            $col_port = '<img src="images/icons/port.png" align="absmiddle"> ' . $port_count . '<br>';
+            $col_port = '<i class="fa fa-link" style="color:#0080FF"></i> ' . $port_count . '<br>';
```
Comment (Rosiak) [inline html/includes/table/devices.inc.php]: Not a fan of the color of this.

File: html/includes/table/devices.inc.php [hunk 2]
```diff
@@ -203,11 +203,11 @@
...
         }
 
         if ($sensor_count) {
-            $col_port .= '<img src="images/icons/sensors.png" align="absmiddle"> ' . $sensor_count;
+            $col_port .= '<i class="fa fa-heartbeat" style="color:#0080FF"></i> ' . $sensor_count;
```
Comment (Rosiak) [inline html/includes/table/devices.inc.php]: Same goes here.

File: html/includes/table/devices.inc.php (hunk 1 above)
Comment (InsaneSplash) [inline html/includes/table/devices.inc.php]: Its easy enough to change... I tried to match the blue icon that was originally used.

File: html/includes/table/devices.inc.php (hunk 2 above)
Comment (InsaneSplash) [inline html/includes/table/devices.inc.php]: Same as above.

**Comment by InsaneSplash** [comment]
I tried to match the current icons that were used for fa in the menus etc, but I agree that it needs some refinement and standardisation. This is something I was going to look in to once I received feedback that this change was interesting enough to pursue.

**Comment by murrant** [comment]
Honestly, we could do with some auditing of icons to make sure they are all consistent and meaningful.  It might even require some icon creation.

**Comment by InsaneSplash** [comment]
Agreed. Would you like me to give it a go and see how it turns out?

**Comment by murrant** [comment]
Sure, but if I were you I'd break it up into manageable steps.

**Comment by InsaneSplash** [comment]
This commit is going to need the latest version of Font Awesome (4.7.0).

**Comment by laf** [comment]
We used to use icons everywhere and have moved a lot to FA.

I'm not personally a fan of using images as icons, it's just extra bloat we don't need. I'm happy with a full switch to FA.

**Comment by laf** [comment]
Looks good to me. @librenms/reviewers

File: html/pages/device/overview/generic/sensor.inc.php [hunk 3]
```diff
@@ -6,12 +6,60 @@
 }
 
 if (count($sensors)) {
+    switch (strtolower($sensor_type)) {
```
Comment (murrant) [inline html/pages/device/overview/generic/sensor.inc.php]: Might want to put this into a function.  There are other places where sensor icons are used, such as the about page.

File: includes/defaults.inc.php [hunk 4]
```diff
@@ -844,6 +844,7 @@
 
 // General GUI options
 $config['gui']['network-map']['style']                  = 'new';//old is also valid
+$config['theme_icon_colour']                            = 'black'; // Font Awesome icon colour customisation
```
Comment (murrant) [inline includes/defaults.inc.php]: This feels odd.  Shouldn't color be mostly dependent on the spot where they are used?
Also, shouldn't this be in CSS, no PHP?

**Comment by murrant** [comment]
Overall, the I like the icon selection 👍 .

There are few spots where color could help out quite a bit.

- The color of the icons on the device overview tabs feels off, I'm not sure if it should match the text or somethings else or if it is fine as is.
- I would really like to seem some colors in the device actions (table view and device overview) to make it easier to identify individual actions.

Those things could be worked out later.  But I do think the default color setting should be moved to css.

**Comment by InsaneSplash** [comment]
Thanks, it's a start in the right direction. I thought about the CSS being
the place to set the icon colour but as users don't generally edit anything
in there it would be easier for them to change the colour of their icon
pallate through a config file than try their hand at the css file which
would probably be replaced through updating.

I also set all the icons to the same colour to try set a standard look. One
has to set a colour as the link style makes them light blue.

I think it will evolve over time but ultimately you guys are in charge :)

**Comment by InsaneSplash** [comment]
cool! 
I'll do a version with the larger icons for you and move the icon colour to the css :)

Lets see what you think?

**Comment by murrant** [comment]
@InsaneSplash `grep -r png .` :-D

**Comment by InsaneSplash** [comment]
eheh yeah, but some need little more investigating :)

_Im done with this PR_. Will review once this has been merged and create a new PR with any changes etc.

**Comment by laf** [comment]
I've squashed this down and rebased for you, contains the agreement message still which is the most important. Just waiting for checks to pass and will merge.

**Pull request merged**